javascript/ql/src/Security/CWE-830/FunctionalityFromUntrustedSource.qhelp

```diff
@@ -0,0 +1,70 @@
+<!DOCTYPE qhelp PUBLIC
+"-//Semmle//qhelp//EN"
+"qhelp.dtd">
+<qhelp>
+	<overview>
+		<p>
+			Including a resource from an untrusted source or using an untrusted channel may
+			allow an attacker to include arbitrary code in the response.
+			When including an external resource (eg., a <code>script</code> element or an
+			<code>iframe</code> element) on a page, it is important to ensure that the received
+			data is not malicious.
+		</p>
+
+		<p>
+			When including external resources, it is possible to verify that the origin (the server
+			that responds to the request) is the intended one by using an <code>https</code> URL.
+			This prevents a MITM (man-in-the-middle) attack where an attacker might have been able
+			to spoof a server response.
+		</p>
+
+		<p>
+			Even when <code>https</code> is used, an attacker might still compromise the origin server.
+			When using a <code>script</code> element, checking for <em>subresource integrity</em>
+			(checking the contents of the data received by supplying a cryptographic digest of the
+			expected sources to the script element) is possible. The script will only load sources
```

Comment:
This sentence could be clearer I think - e.g.

```suggestion
			When you use a <code>script</code> element, you can check for subresource integrity - that is, you can check the contents of the data received by supplying a cryptographic digest of the
			expected sources to the <code>script</code> element. The script will only load sources
```

I don't think we need "subresource integrity" to be italicized.